virttest/utils_test/__init__.py

```diff
@@ -718,6 +718,165 @@ def transfer_data(session, host_cmd, guest_cmd, n_time, timeout,
+                success = False
+                continue
+            try:
+                mux = test[1]
+            except KeyError:
+                pass
+            cmd = "avocado run %s --job-results-dir %s --job-timeout %d" % (testcase, result_path, timeout)
+            if mux:
+                cmd += " -m %s" % mux
+            if add_args:
+                cmd += " %s" % add_args
+            if session.cmd_status(cmd, timeout=timeout) > 0:
+                logging.error("Testcase: %s has failures consult "
+                              "the logs for details\nstatus: %s", testcase, status)
+                success = False
+        return success
+
+    def get_results(test_path, result_path, session):
+        """
+        Copy avocado results present on the guest back to the host.
+        """
+        logging.debug("Trying to copy avocado results from guest")
+        cmd = "[ -n \"$(ls %s)\" ]" % result_path
+        if session.cmd_status(cmd) > 0:
+            logging.error("No results folder found")
+            return False
+        guest_results_dir = utils_misc.get_path(test.debugdir, "guest_avocado_results")
+        os.makedirs(guest_results_dir)
+        # result info tarball to host result dir
+        results_tarball = os.path.join(test_path, "results.tgz")
+        compress_cmd = "cd %s && " % result_path
+        compress_cmd += "tar cjvf %s ./*" % results_tarball
+        compress_cmd += " --exclude=*core*"
+        compress_cmd += " --exclude=*crash*"
+        session.cmd(compress_cmd, timeout=600)
+        vm.copy_files_from(results_tarball, guest_results_dir)
+        # cleanup results dir from guest
+        clean_cmd = "rm -f %s;rm -rf %s" % (results_tarball, result_path)
+        session.cmd(clean_cmd, timeout=240)
+        results_tarball = os.path.basename(results_tarball)
+        results_tarball = os.path.join(guest_results_dir, results_tarball)
+        uncompress_cmd = "tar xjvf %s -C %s" % (results_tarball,
+                                                guest_results_dir)
+        process.run(uncompress_cmd)
+        process.run("rm -f %s" % results_tarball)
+        return True
+
+    # Run test
+    test_path = params.get("vm_test_path", "/var/tmp/avocado/")
+    result_path = os.path.join(test_path, "results")
+    if not os.path.isdir(test_path):
+        os.makedirs(test_path)
+    if not os.path.isdir(result_path):
+        os.makedirs(result_path)
+    if not install_avocado(installtype):
+        test.error("avocado installation failed, consult previous errors")
+
+    test_status = runtest(session, testrepo, testlist, timeout, result_path, test_path)
+    guest_result_path = get_results(test_path, result_path, session)
+    # Cleanup
+    session.cmd_status("rm -rf %s" % test_path)
+    if (not test_status or not guest_result_path) and not ignore_result:
+        test.fail("consult previous errors")
+    else:
+        return (test_status or guest_result_path)
```

Comment:
Sure, reason for this consider both results none and test fail as a failure return, will check how better to handle both separately.